gobblin-scheduler/src/main/java/gobblin/scheduler/SchedulerDaemon.java

```diff
@@ -72,20 +72,22 @@ public SchedulerDaemon(Properties defaultProperties, Properties customProperties
     List<Service> services = Lists.<Service>newArrayList(new JobScheduler(properties));
     boolean jobExecInfoServerEnabled = Boolean
         .valueOf(properties.getProperty(ConfigurationKeys.JOB_EXECINFO_SERVER_ENABLED_KEY, Boolean.FALSE.toString()));
+    boolean adminUiServerEnabled = Boolean
+        .valueOf(properties.getProperty(ConfigurationKeys.ADMIN_SERVER_ENABLED_KEY, Boolean.FALSE.toString()));
     if (jobExecInfoServerEnabled) {
+      LOG.info("Starting the job execution info server since it is enabled");
       JobExecutionInfoServer executionInfoServer = new JobExecutionInfoServer(properties);
       services.add(executionInfoServer);
-      if (shouldRunAdminServer(properties)) {
+      if (adminUiServerEnabled) {
```

Comment:
That is true.  I would really like to get rid of the duplication between SchedulerDaemon and GobblinYarnApplicationLauncher.  I considered wrapping the admin server creation code in a factory, but it is a little harder because it relies on the execution info server being configured.